competition-simulator/programming/time.md

```diff
@@ -1,62 +1,23 @@
 ---
 layout: page
-title: Simulated Time
+title: Robot Time
 ---
 
-# Simulated Time
+# Important methods
 
 In the simulated environment, time advances only at the pace that the simulator
 is run and in cooperation with the controller code (such as the code controlling
-your robot). As a result, using `time.time` to know how long your robot has been
-running for or `time.sleep` to wait for some duration may be unreliable.
+your robot). As a result, using the `time` method to know how long your robot has been
+running for or `sleep` method to wait for some duration is the most reliable way of 
+implementing precise movements in your robot
 
-## Approaches
+<div class="info">    
+  If you find that the simulator freezes then this indicates that your code is    
+  busy doing something without giving the simulator a chance to run.    
 
-### Existing projects: the `Robot` class
-
-<!-- We should drop this after SR2020 and move `ManualTimestepRobot` to being the default. -->
-
-For existing projects the default `Robot` class will automatically advance the
-simulator all the time. This can make programming your robot slightly easier as
-you don't need to worry about the progress of time and (just like a real robot)
-you can assume that time passing will just happen.
-
-However this has a drawback -- because the simulator advances time in small
-chunks (rather than the smooth progression we're used to in reality) it can mean
-that your robots actions sometimes run for slightly more or less time than you
-expect. While the time difference will be small (a few tens of milliseconds), it
-is likely to impact attempts at more precise movement more than larger actions
-due to their shorter time.
-
-This may mean for example that turning by a small angle to line up with a token
-will sometimes work and sometimes point the robot in not quite the right
-direction.
-
-If your robot code is impacted by these unpredictability issues we recommend
-that you change over to using the `ManualTimestepRobot` class instead. For
-guidance on doing this, see the [upgrade guide](#upgrading) below.
-
-This class is maintained for compatibility with earlier releases of the
-simulator, though its use is discouraged (especially for new projects).
-
-### New projects: the `ManualTimestepRobot` class {#manual-timestep-robot}
-
-For new projects, or existing projects that want to be sure of getting precise
-robot behaviour, the recommended approach is to use the `ManualTimestepRobot`
-class.
-
-This approach relies upon your code advancing the simulation explicitly by
-calling its `sleep` method (documented below) in order for the simulation to
-actually run. This should not be an issue for most robot code however as you
-will likely be doing this anyway in order to wait for things to happen.
-
```

Comment:
I think we probably do want to keep this paragraph or something like it. While we do allude to this elsewhere, I think the clear idea that this (especially the first sentence) gives -- that the competitors code is directly responsible for the advancement of time and that they can break things if they get it wrong is really important.

(It also sets up the context for the following info box fairly well)

Reply:
b126b3c